deepspeed/utils/timer.py

```diff
@@ -42,25 +42,40 @@ def __init__(self, name):
             self.event_timers = []
             self.start_event = None
             self.elapsed_records = None
+            self.start_time = 0.0
```

Comment:
I think it would be cleaner to implement separate timers for synchronized devices (i.e., CPU) and asynchronous devices (i.e., GPU). This will help to avoid the repeated `is_synchronized_device()` calls and reduce the timer overhead.

Reply:
Seperate timer will require caller to treat synchronized device and asynchronous device differently.  By looking at a typical caller like this link(https://github.com/microsoft/DeepSpeed/blob/master/tests/onebit/test_mpi_perf.py#L56),  keep caller not distinguish between two kinds of device is still possible.

An alternative could be call `is_synchronized_device()` once in `init` function and store the result, then use that result to avoid repeatedly function call.

Reply:
Changed to call is_synchronized_device() only once during timer init.